DigitalExcellence/dex-backend
# Feature/217 project icon

## Description

Added:
 - FileController, FileService, FileRepository, FileModel
 - Extension: FileUploader which is responsible for handling uploads to the file server.
 - FileResourceResult, FileResource
 - AutoMapper for FileModel and FileResourceResult
 - Folder: files which is for storing files.
 - Dependency injection for FileRepository and FileService
 - Migrations for Database
- AllowedExtensionAttribute and MaxFileSizeAttribute

Updated:
 - .gitignore now ignores content of files folder
 - ProjectController now also accepts a reference to file (project-icon) on CreateProject and UpdateProject
 - Project Model (and ProjectResourceResult) now also has a file field (project-icon)

## Type of change


## Checklist


## Steps to Test or Reproduce

1. Run API and Identity server
2. Run runtests_windows.bat or runtests_linux.sh. (depending on your OS)
3. Check if all tests are running
4. Run UploadFile endpoint manually.
5. Check if file was uploaded to "API/wwwroot/Resources"

## Link to issue

Closes: #217
